sympy/integrals/manualintegrate.py

```diff
@@ -254,16 +271,12 @@ def possible_subterms(term):
                 r.extend(possible_subterms(u))
             return r
         elif isinstance(term, Pow):
-            r = []
-            if term.args[1].is_constant(symbol):
-                r.append(term.args[0])
-            elif term.args[0].is_constant(symbol):
-                r.append(term.args[1])
-            if term.args[1].is_Integer:
-                r.extend([term.args[0]**d for d in divisors(term.args[1])
+            r = [arg for arg in term.args if arg.has(symbol)]
+            if term.exp.is_Integer:
+                r.extend([term.base**d for d in primefactors(term.exp)
```

Comment:
Although prime factors is better than divisors it can still be very slow for large integers. Also presumably we want to limit how many possible subterms are generated to avoid testing them all.

Why is it necessary to consider the prime factors here? What is this code used for?

I guess this is for something like this:
```python
In [5]: i = Integral(x**(n-1)/(1 + x**n), x)

In [6]: i
Out[6]: 
⌠          
⎮  n - 1   
⎮ x        
⎮ ────── dx
⎮  n       
⎮ x  + 1   
⌡          

In [7]: i.subs(n, 2*3*5*7*11).integrate(manual=True)
^C---------------------------------------------------------------------------
KeyboardInterrupt
```
Do we need to consider all prime factors? Is it equivalent to considering all divisors? Is there a more direct way to determine what are good powers to try (rather than brute force)?

Reply:
This example hangs here, master and 3.10. It's not because `primefactors` provides too many possibilities. Actually substitution rule succeeds, but `partial_fractions_rule` hangs.
```py
(x**(n-1)/(1 + x**n)).subs(n, 2*3*5*7*11).apart()
```

Reply:
I realise that it is apart that hangs but why doesn't it just use the substitution that is found here? Presumably the purpose of the `possible_subterms` function is to identify this.

Reply:
Because `AlternativeRule` are meant to provide different methods. Even a previous rule succeeds, it tries other kinds of rules.
https://github.com/sympy/sympy/blob/ce53e7b5aeee7019cf019b9e7cbea07fe3f8cbac/sympy/integrals/manualintegrate.py#L338-L345
I also find it annoying sometimes. Perhaps we should add a flag to disable it by default, so it returns immediately when a rule succeeds.

Reply:
What is the advantage of trying other rules after a previous one succeeds?

Reply:
The only advantage is to provide different kinds of steps for users.
See a bad example https://sympy-beta.vercel.app/input/integrate((x+2)%2F((x+3)*(x-4)))
I'm open to remove `AlternativeRule` completely.